### ISSUE TITLE
logs persistency failed if enabled

### ISSUE THREAD

**Issue by Gmasquelier** [airflow-helm/charts#113]
**chart version 8.0.2** 

It looks like we're not able to create log persistency as the templates helpers for logs is providing mountpath and subpath parameters for volumes in the spec.

https://github.com/airflow-helm/charts/blob/8611c766992a9f7156aad450f17beb2213cf667c/charts/airflow/templates/_helpers/pods.tpl#L223

Thanks for your work on this new version otherwise.

**Comment by murilommen** [comment]
Having the same issue here! even though my Values.logs.path is not empty, it returns me this:
`Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(Deployment.spec.template.spec.volumes[1]): unknown field "mountPath" in io.k8s.api.core.v1.Volume`

**Comment by karakanb** [comment]
I am having the same issue with the 8.0.2 release. Here's my values for reference:
```yaml
logs:
    path: /opt/airflow/logs
    persistence:
        enabled: false
        storageClass: nfs
```

**Comment by commandosss** [comment]
confirmed. i'm having the same

**Comment by kirandeshpande1990** [comment]
Yes when will this be fixed? we have a release next week where I need this chart to be working with persistent storage

**Comment by karakanb** [comment]
@thesuperzapper I might be able to contribute to this, but I am not really sure how to fix it without changing the way volumes are reused throughout the entire chart. do you have a quick solution in mind that'd unblock this for everyone? if you don't have the time but can give some pointers, I might try to prepare a PR.

currently, the chart doesn't work with persistent storage for logs.

**Comment by commandosss** [comment]
@karakanb 
Fix for templates/_helper/pods.tpl
U need to change 220-225 to the following: 
```
{{- /* logs */ -}}
{{- if .Values.logs.persistence.enabled }}
- name: logs-data
  persistentVolumeClaim:
    {{- if .Values.logs.persistence.existingClaim }}
    claimName: {{ .Values.logs.persistence.existingClaim }}
    {{- else }}
    claimName: {{ printf "%s-logs" (include "airflow.fullname" . | trunc 58) }}
    {{- end }}
{{- end }}

**Comment by karakanb** [comment]
I am not sure if that'd fix the main problem because I believe these helper parts are used for both `volumeMounts` as well as `volumes` definitions on the deployments, therefore there'd need to be some sort of a conditional rendering of the `mountPath` since it doesn't seem to be valid for the `volumes` schema, although I am not very confident about what I say here, so take it with a grain of salt.

Would you mind submitting a PR with this change so that we could save some time if it is the right approach?

**Comment by commandosss** [comment]
@karakanb i'm so sorry, but i don't have a time on that

**Comment by karakanb** [comment]
I have created a PR with this change, let's see how it goes.

**Comment by commandosss** [comment]
@karakanb your change is wrong
u need to change lines 220-225
Thanks for the PR

**Comment by karakanb** [comment]
I updated the PR to reflect the correct changes.

**Comment by thesuperzapper** [comment]
@Gmasquelier @commandosss can you confirm if this has been fixed after version `8.0.6` of the chart?
